tests/test_run.py

```diff
@@ -768,3 +768,25 @@ def test(self):
         self.assertEqual(ret, 0)
         self.assertTrue(os.path.isfile(fname))
         self.assertEqual(len(os.listdir(self.dvc.cache.local.cache_dir)), 1)
+
+
+class TestRunMarksRemove(TestDvc):
+    def test(self):
+        file = "file.txt"
+        file_content = "content"
+        stage_file = file + Stage.STAGE_FILE_SUFFIX
+
+        ret = main(
+            [
+                "run",
+                "-o",
+                file,
+                "--outs-no-remove",
```

Comment:
`--outs-no-remove` should work the same as existing `--outs` and `--outs-no-cache`.

Reply:
So `dvc run --outs-no-remove foo` should be used and not `dvc run --outs foo --outs-no-remove foo`. I like the idea of actually marking them post-factum, but for now we should stay in sync with existing behaviour.